### PR TITLE
A new CLI based on stacks.yml

### PR DESCRIPTION
**Pull request by allait**
New create/delete/provision/deploy commands interface:

    dmaws <stage> <environment> <command> [apps] [command options and arguments]

`create`/`delete` commands use `stacks.yml` for CloudFormation stacks instead of Ansible.
`deploy` command creates an archive with repository contents and build artefacts

### CODE REVIEW

File: dmaws/build.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    run_git_cmd(['archive', '--format=zip', 'HEAD'], cwd, stdout=package_file)
+
+    os.close(package_file)
+
+    return ref, sha, file_path
+
+
+def run_project_build_script(cwd):
+    run_cmd(['./scripts/build.sh'], cwd=cwd)
```
Comment (TheDoubleK) [inline dmaws/build.py]: Where is this `build.sh` script? Can't see it in this project or in the apps being deployed?

Comment (allait) [inline dmaws/build.py]: It's run from the repository being deployed. I have a script for the admin-frontend that I'll send for review a bit later.

**Comment by TheDoubleK** [comment]
It would be good to have the instructions documented in the README as well as the pull request message, just to keep it current.

**Comment by allait** [comment]
@TheDoubleK there's `--help` for the CLI that has a full list of commands and accepted arguments/options for each one. I'll look into adding it to README once we're happy with the CLI.

**Comment by TheDoubleK** [comment]
OK fair enough.

File: dmaws/commands/provision.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import json
+import subprocess
+
+from ..cli import cli_command
+from ..utils import run_cmd
+
+
+def run_playbook(playbook, hosts, ctx, basedir='playbooks/'):
+    args = [
+        'ansible-playbook',
```
Comment (TheDoubleK) [inline dmaws/commands/provision.py]: So (just to clarify) do we still need Ansible for 'provision' but are dropping it for create/delete/deploy?

File: vars/production.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+subnets:
+  - subnet-bf8c5ce6
+  - subnet-916bfcf4
+  - subnet-82dc63f5
```
Comment (TheDoubleK) [inline vars/production.yml]: The `production.yml` and `preview.yml` subnets are identical - is that right? And if so do we need both files, or could we just have one file that is used for both stages?

File: dmaws/commands/provision.py (hunk 2 above)
Comment (allait) [inline dmaws/commands/provision.py]: Yeah. And we're gathering the variables in python and sending the full dictionary to ansible, so it's only responsible for actually running the provisioning tasks

**Pull request merged**